HKMP/Game/Client/PlayerManager.cs

```diff
@@ -175,29 +301,101 @@ internal class PlayerManager {
+        public void RecycleAllPlayers() {
+            foreach (var (id, playerData) in _playerData) {
+                // Recycle player
+                RecyclePlayer(id);
+            }
+        }
+
+        /// <summary>
+        /// Reset the player with the given ID to its initial state.
+        /// </summary>
+        /// <param name="id">The player ID.</param>
+        public void ResetPlayer(ushort id) {
+            if (!_playerData.TryGetValue(id, out var playerData)) {
+                Logger.Get().Warn(this, $"Tried to reset player that does not exists for ID {id}");
+                return;
+            }
+
+            var container = playerData.PlayerContainer;
+            if (container is null) return;
+
+            // Destroy all descendants and components that weren't originally on the container object.
+            foreach (Transform child in container.transform) {
+                switch (child.name) {
+                    case "PlayerPrefab":
+                        foreach (var component in child.GetComponents<Component>()) {
+                            if (!_prefabComponentTypes.Contains(component.GetType())) {
+                                Object.Destroy(component);
+                            }
+                        }
+
+                        foreach (Transform grandChild in child) {
+                            if (grandChild.name is "Attacks" or "Effects" or "Spells") {
+                                foreach (var component in grandChild.GetComponents<Component>()) {
+                                    Object.Destroy(component);
+                                }
+                                
+                                foreach (Transform greatGrandChild in grandChild) {
+                                    Object.Destroy(greatGrandChild.gameObject);
+                                }
+                            } else {
+                                Object.Destroy(grandChild.gameObject);
+                            }
+                        }
+
+                        break;
+                    case "Username":
+                        foreach (var component in child.GetComponents<Component>()) {
+                            if (!_usernameComponentTypes.Contains(component.GetType())) {
+                                Object.Destroy(component);
+                            }
+                        }
+
+                        foreach (Transform grandChild in child) {
+                            Object.Destroy(grandChild.gameObject);
+                        }
+
+                        break;
+                    default:
+                        Object.Destroy(child);
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Reset all existing players.
+        /// </summary>
+        public void ResetAllPlayers()
+        {
+            foreach (var (id, playerData) in _playerData)
+            {
```

Comment:
Style change and you are not using one of the variables from the deconstruction.
```suggestion
            foreach (var (id, _) in _playerData) {
```